### DISCUSSION TITLE
stochfish engine using CPU (maybe the wrong place to ask but maybe)

### DISCUSSION THREAD

**Discussion by m342** [thomas-mauran/chess-tui#115] [General]
Hi i m using ~/.cargo/bin/chess-tui -e /usr/bin/stockfish  

https://github.com/thomas-mauran/chess-tui

and i see cpu average acivity as i m playing as an humain again stockfish 
even when it is my turn to play  (i play as (me) human vs bot).
is it relevant to stockfish or chess-tui ? 

as I am not a dev, just end user i ask here as i don't know how it worls. 

thanks and sorry if it's not relevant to this project.

**Comment by thomas-mauran** [comment]
Hello @m342 I'm not sure to see the problem, maybe you can add a screenshot to showcase the situation ?

**Reply by m342** [reply]
Hi (bonjour), thanks for you reply 

not much consumption but if i launch top / htop i see this even it's my turn to play. not a big deal but i was asking myself and as i  said i do not know how it works -> but why cpu at this time ? maybe the engine who try to work when i m note here :) ? 

![image](https://github.com/user-attachments/assets/35af4a5b-f31a-4683-a3ff-30ee17e2fa3a)


by the way thanks for you work.

**Reply by mfernd** [reply]
Is the view mode on "Tree" to view subprocesses?

**Reply by thomas-mauran** [reply]
Hello @m342,

This is an interesting observation. I reproduced the screenshot by also using htop, and here's what I found:

- When playing in solo mode (and spamming), the chess-tui process itself consumes around 7-10% of the CPU.
- When playing against a chess engine, the CPU usage increases due to both chess-tui and the Stockfish engine processes. Every time the engine needs to think of a move, Stockfish is spawned, and you can see its process pop up in htop, consuming additional CPU resources.

Currently, the game only spawns a new Stockfish process when requesting the bot's move. Outside of that, Stockfish isn't being called. As a result, the 5% CPU usage you see is primarily due to the chess-tui process itself. This is because the game uses a loop to constantly render the board at a certain tick rate, which is resource-intensive.

Tell me if you have any other questions about the game and thanks for your support :)
